impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js

```diff
@@ -409,17 +410,24 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dij
   _updatePicklistAvailableValues: function(values) {
     construct.empty(this.picklistAvailableValues.domNode);
     array.forEach(values, function (result, idx) {
-      if (Array.isArray(result)) {
-        result.forEach((val) => val.replace(/&amp;/g, '&'));
-      } else if (typeof myVar === 'string' || myVar instanceof String) {
-        result.replace(/&amp;/g, '&');
-      }
+      result = this._unescapeAmpIfExist(result);
       this.containerNode.options[idx] = new Option(result, result);
       this.containerNode.options[idx].title = result;
     }, this.picklistAvailableValues);
     this.picklistLoaded = true;
   },
 
+  _unescapeAmpIfExist: function(str) {
+    if (Array.isArray(str)) {
+      str.forEach(function(val) {
+        val.replace(/&amp;/g, '&'));
```

Comment:
@PashkovichAndrei extra `)` here. Sorry I missed it... wingman didn't caught it either.